modules/folder/main.tf

```
@@ -48,3 +62,30 @@ resource "google_compute_firewall_policy_association" "default" {
   name              = var.firewall_policy.name
   firewall_policy   = var.firewall_policy.policy
 }
+
+resource "google_assured_workloads_workload" "folder" {
+  count                     = var.assured_workload_config == null ? 0 : 1
```

```suggestion
  count                     = (var.folder_create && var.assured_workload_config != null) ? 1 : 0
```
I guess that it doesn't make sense to have `assured_workload_config` and `folder_create == false` as we can't "upgrade" existing folder to Assured Workload folder? Maybe it's worth surfacing that in the README.md?

---

Done, good catch